File: src/backend/cte/src/model/cte_perigoso_model.py
from src import db


class CtePerigosoModel(db.Model):
    __tablename__ = 'cte_perigoso'

    id = db.Column(db.Integer, primary_key=True)
    numero_onu = db.Column(db.String(4))
    nome_apropriado = db.Column(db.String(150))
    classe_risco = db.Column(db.String(40))
    grupo_embalagem = db.Column(db.String(6))
    quantidade_total_produto = db.Column(db.String(20))
    quantidade_tipo_volume = db.Column(db.String(60))
    ponto_fulgor = db.Column(db.String(6))
    id_cte_cabecalho = db.Column(db.Integer, db.ForeignKey('cte_cabecalho.id'))


    def mapping(self, data):
        self.id = data.get('id')
        self.id_cte_cabecalho = data.get('idCteCabecalho')
        self.numero_onu = data.get('numeroOnu')
        self.nome_apropriado = data.get('nomeApropriado')
        self.classe_risco = data.get('classeRisco')
        self.grupo_embalagem = data.get('grupoEmbalagem')
        self.quantidade_total_produto = data.get('quantidadeTotalProduto')
        self.quantidade_tipo_volume = data.get('quantidadeTipoVolume')
        self.ponto_fulgor = data.get('pontoFulgor')

    def serialize(self):
        return {
            'id': self.id,
            'idCteCabecalho': self.id_cte_cabecalho,
            'numeroOnu': self.numero_onu,
            'nomeApropriado': self.nome_apropriado,
            'classeRisco': self.classe_risco,
            'grupoEmbalagem': self.grupo_embalagem,
            'quantidadeTotalProduto': self.quantidade_total_produto,
            'quantidadeTipoVolume': self.quantidade_tipo_volume,
            'pontoFulgor': self.ponto_fulgor,
        }

File: src/frontend/cte/lib/app/data/model/cte_perigoso_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';


class CtePerigosoModel {
	int? id;
	int? idCteCabecalho;
	String? numeroOnu;
	String? nomeApropriado;
	String? classeRisco;
	String? grupoEmbalagem;
	String? quantidadeTotalProduto;
	String? quantidadeTipoVolume;
	String? pontoFulgor;

	CtePerigosoModel({
		this.id,
		this.idCteCabecalho,
		this.numeroOnu,
		this.nomeApropriado,
		this.classeRisco,
		this.grupoEmbalagem,
		this.quantidadeTotalProduto,
		this.quantidadeTipoVolume,
		this.pontoFulgor,
	});

	static List<String> dbColumns = <String>[
		'id',
		'numero_onu',
		'nome_apropriado',
		'classe_risco',
		'grupo_embalagem',
		'quantidade_total_produto',
		'quantidade_tipo_volume',
		'ponto_fulgor',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Numero Onu',
		'Nome Apropriado',
		'Classe Risco',
		'Grupo Embalagem',
		'Quantidade Total Produto',
		'Quantidade Tipo Volume',
		'Ponto Fulgor',
	];

	CtePerigosoModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		idCteCabecalho = jsonData['idCteCabecalho'];
		numeroOnu = jsonData['numeroOnu'];
		nomeApropriado = jsonData['nomeApropriado'];
		classeRisco = jsonData['classeRisco'];
		grupoEmbalagem = jsonData['grupoEmbalagem'];
		quantidadeTotalProduto = jsonData['quantidadeTotalProduto'];
		quantidadeTipoVolume = jsonData['quantidadeTipoVolume'];
		pontoFulgor = jsonData['pontoFulgor'];
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['idCteCabecalho'] = idCteCabecalho != 0 ? idCteCabecalho : null;
		jsonData['numeroOnu'] = numeroOnu;
		jsonData['nomeApropriado'] = nomeApropriado;
		jsonData['classeRisco'] = classeRisco;
		jsonData['grupoEmbalagem'] = grupoEmbalagem;
		jsonData['quantidadeTotalProduto'] = quantidadeTotalProduto;
		jsonData['quantidadeTipoVolume'] = quantidadeTipoVolume;
		jsonData['pontoFulgor'] = pontoFulgor;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		idCteCabecalho = plutoRow.cells['idCteCabecalho']?.value;
		numeroOnu = plutoRow.cells['numeroOnu']?.value;
		nomeApropriado = plutoRow.cells['nomeApropriado']?.value;
		classeRisco = plutoRow.cells['classeRisco']?.value;
		grupoEmbalagem = plutoRow.cells['grupoEmbalagem']?.value;
		quantidadeTotalProduto = plutoRow.cells['quantidadeTotalProduto']?.value;
		quantidadeTipoVolume = plutoRow.cells['quantidadeTipoVolume']?.value;
		pontoFulgor = plutoRow.cells['pontoFulgor']?.value;
	}	

	CtePerigosoModel clone() {
		return CtePerigosoModel(
			id: id,
			idCteCabecalho: idCteCabecalho,
			numeroOnu: numeroOnu,
			nomeApropriado: nomeApropriado,
			classeRisco: classeRisco,
			grupoEmbalagem: grupoEmbalagem,
			quantidadeTotalProduto: quantidadeTotalProduto,
			quantidadeTipoVolume: quantidadeTipoVolume,
			pontoFulgor: pontoFulgor,
		);			
	}

	
}